XPathUnhandledException on export of forms using 3rd party client functions

#### Software versions 
- Briefcase v1.16.3
- Java v1.8.x
- Windows 10

#### Problem description
When doing an export, `XPathUnhandledException` occurs, causing Briefcase to hang indefinitely.

#### Steps to reproduce the problem
1. **XLSForm**: Convert to XML an XLSForm that has a custom functionality intended to be used by a 3rd party client, i.e. not ODK Collect, e.g. [JHU Collect](https://github.com/PMA-2020/collect).
2. **Aggregate**: Upload form def to aggregate
3. **Client**: Download, complete, and submit form using a 3rd party client
4. **Briefcase**: Load GUI via the CLI
5. **Briefcase**: Fetch submissions
6. **Briefcase**: Attempt to export
7. **Briefcase**: Observe error stacktrace in terminal window

#### Expected behavior
- When exporting, should not error / freeze / crash.
- Briefcase should only care about the submission data; should not care about implementation details of how that data was created.

#### Other information 
**The particular issue**
The JHU Collect feature causing this issue is `trim()`, which clears extraneous whitespace. I believe that ODK Collect now supports this in a somewhat new feature, `normalize-space()`. In any event, I suppose it doesn't matter the specific 3rd party feature that was used. I'm just thinking that it would be ideal if Briefcase did not fully parse / evaluate / validate the form definition during the export (or any) process, unless there is a need for it that I am not seeing.

**Stack trace**
```java
C:\Users\annro>java -jar Downloads\ODK-Briefcase-v1.16.3.jar
Exception in thread "Thread-4" java.lang.RuntimeException: Error evaluating field 'firstname': The problem was located in calculate expression for /FSQ/firstname
XPath evaluation: cannot handle function 'trim'
       at org.javarosa.core.model.LatestDagBase.evaluateTriggerable(LatestDagBase.java:139)
       at org.javarosa.core.model.LatestDagBase.doEvaluateTriggerables(LatestDagBase.java:63)
       at org.javarosa.core.model.Safe2014DagImpl.evaluateTriggerables(Safe2014DagImpl.java:366)
       at org.javarosa.core.model.Safe2014DagImpl.initializeTriggerables(Safe2014DagImpl.java:452)
       at org.javarosa.core.model.Safe2014DagImpl.initializeTriggerables(Safe2014DagImpl.java:430)
       at org.javarosa.core.model.FormDef.initializeTriggerables(FormDef.java:721)
       at org.javarosa.core.model.FormDef.initialize(FormDef.java:1306)
       at org.opendatakit.briefcase.export.FormDefinition.getFormControls(FormDefinition.java:148)
       at org.opendatakit.briefcase.export.FormDefinition.from(FormDefinition.java:142)
       at org.opendatakit.briefcase.ui.export.ExportPanel.lambda$export$10(ExportPanel.java:190)
       at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
       at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
       at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
       at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
       at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
       at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
       at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
       at java.util.stream.ReferencePipeline.forEach(Unknown Source)
       at org.opendatakit.briefcase.reused.job.JobsRunner.launchAsync(JobsRunner.java:63)
       at org.opendatakit.briefcase.reused.job.JobsRunner.launchAsync(JobsRunner.java:51)
       at org.opendatakit.briefcase.ui.export.ExportPanel.export(ExportPanel.java:217)
       at java.lang.Thread.run(Unknown Source)
Caused by: org.javarosa.xpath.XPathUnhandledException: The problem was located in calculate expression for /FSQ/firstname
XPath evaluation: cannot handle function 'trim'
       at org.javarosa.xpath.expr.XPathFuncExpr.eval(XPathFuncExpr.java:486)
       at org.javarosa.xpath.XPathConditional.evalRaw(XPathConditional.java:71)
       at org.javarosa.core.model.condition.Triggerable.evalRaw(Triggerable.java:219)
       at org.javarosa.core.model.condition.Recalculate.eval(Recalculate.java:63)
       at org.javarosa.core.model.condition.Triggerable.apply(Triggerable.java:152)
       at org.javarosa.core.model.LatestDagBase.evaluateTriggerable(LatestDagBase.java:125)
       ... 21 more
```

**Versions confirmed to work / not work w/ 3rd party xpath functions**
Working: 1.9.x
Not working: 1.16.3, 1.15.x

--- a/build.gradle
+++ b/build.gradle
@@ -42,7 +42,7 @@ dependencies {
     exclude group: 'commons-logging'
   }
   compile group: 'net.sf.kxml', name: 'kxml2', version: '2.3.0'
-  compile(group: 'org.opendatakit', name: 'opendatakit-javarosa', version: '2.15.1') {
+  compile(group: 'org.opendatakit', name: 'opendatakit-javarosa', version: '2.16.0') {
     exclude group: 'org.slf4j'
   }
   compile group: 'org.hsqldb', name: 'hsqldb', version: '2.4.0'

--- a/src/org/opendatakit/briefcase/export/ExportConfiguration.java
+++ b/src/org/opendatakit/briefcase/export/ExportConfiguration.java
@@ -80,7 +80,7 @@ public class ExportConfiguration {
   private final OverridableBoolean removeGroupNames;
   private final OverridableBoolean smartAppend;
 
-  private ExportConfiguration(Optional<String> exportFileName, Optional<Path> exportDir, Optional<Path> pemFile, DateRange dateRange, OverridableBoolean pullBefore, OverridableBoolean overwriteFiles, OverridableBoolean exportMedia, OverridableBoolean splitSelectMultiples, OverridableBoolean includeGeoJsonExport, OverridableBoolean removeGroupNames, OverridableBoolean smartAppend) {
+  public ExportConfiguration(Optional<String> exportFileName, Optional<Path> exportDir, Optional<Path> pemFile, DateRange dateRange, OverridableBoolean pullBefore, OverridableBoolean overwriteFiles, OverridableBoolean exportMedia, OverridableBoolean splitSelectMultiples, OverridableBoolean includeGeoJsonExport, OverridableBoolean removeGroupNames, OverridableBoolean smartAppend) {
     this.exportFileName = exportFileName;
     this.exportDir = exportDir;
     this.pemFile = pemFile;

--- a/src/org/opendatakit/briefcase/export/FormDefinition.java
+++ b/src/org/opendatakit/briefcase/export/FormDefinition.java
@@ -37,8 +37,6 @@ import org.javarosa.core.model.IDataReference;
 import org.javarosa.core.model.IFormElement;
 import org.javarosa.core.model.ItemsetBinding;
 import org.javarosa.core.model.QuestionDef;
-import org.javarosa.core.model.condition.EvaluationContext;
-import org.javarosa.core.model.condition.IFunctionHandler;
 import org.javarosa.core.model.instance.DataInstance;
 import org.javarosa.core.model.instance.ExternalDataInstance;
 import org.javarosa.core.model.instance.InstanceInitializationFactory;
@@ -53,32 +51,6 @@ import org.opendatakit.briefcase.reused.BriefcaseException;
  * This class holds all the relevant information about the form being exported.
  */
 public class FormDefinition {
-  private static final IFunctionHandler DUMMY_PULLDATA_HANDLER = new IFunctionHandler() {
-    @Override
-    public String getName() {
-      return "pulldata";
-    }
-
-    @Override
-    public List<Class[]> getPrototypes() {
-      return Collections.emptyList();
-    }
-
-    @Override
-    public boolean rawArgs() {
-      return true;
-    }
-
-    @Override
-    public boolean realTime() {
-      return false;
-    }
-
-    @Override
-    public Object eval(Object[] args, EvaluationContext ec) {
-      return "";
-    }
-  };
   private final String id;
   private final String name;
   private final Path formFile;
@@ -144,7 +116,7 @@ public class FormDefinition {
   }
 
   private static Map<String, QuestionDef> getFormControls(FormDef formDef) {
-    formDef.getEvaluationContext().addFunctionHandler(DUMMY_PULLDATA_HANDLER);
+    formDef.getEvaluationContext().addFallbackFunctionHandler((name, args, ec) -> "");
     formDef.initialize(false, new InstanceInitializationFactory());
     return formDef.getChildren()
         .stream()

--- a/test/java/org/opendatakit/briefcase/CustomFunctionTest.java
+++ b/test/java/org/opendatakit/briefcase/CustomFunctionTest.java
@@ -1,26 +1,15 @@
 package org.opendatakit.briefcase;
 
-import static org.opendatakit.briefcase.reused.UncheckedFiles.createDirectories;
-import static org.opendatakit.briefcase.reused.UncheckedFiles.createTempDirectory;
+import static org.junit.Assert.assertThat;
 import static org.opendatakit.briefcase.reused.UncheckedFiles.toURI;
 
-import java.io.IOException;
-import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Optional;
+import org.hamcrest.Matchers;
 import org.junit.Test;
-import org.opendatakit.briefcase.export.DateRange;
-import org.opendatakit.briefcase.export.ExportConfiguration;
-import org.opendatakit.briefcase.export.ExportToCsv;
 import org.opendatakit.briefcase.export.FormDefinition;
-import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
-import org.opendatakit.briefcase.model.FormStatus;
-import org.opendatakit.briefcase.model.form.FormMetadata;
-import org.opendatakit.briefcase.model.form.InMemoryFormMetadataAdapter;
 import org.opendatakit.briefcase.pull.aggregate.PullFromAggregateIntegrationTest;
-import org.opendatakit.briefcase.reused.OverridableBoolean;
-import org.opendatakit.briefcase.util.BadFormDefinition;
 
 public class CustomFunctionTest {
   private static Path getPath(String fileName) {
@@ -31,37 +20,9 @@ public class CustomFunctionTest {
 
   @Test
   public void name() {
-    FormDefinition.from(getPath("custom-function-form.xml"));
-  }
-
-  @Test
-  public void full_regression_test() throws IOException, BadFormDefinition {
-    Path exportDir = createTempDirectory("briefcase-test-export-dir-");
-    Path sourceFormFile = getPath("custom-function-form.xml");
-    Path briefcaseDir = createTempDirectory("briefcase-test-storage-dir-");
-    Path formFile = briefcaseDir.resolve("Custom function form").resolve("Custom function form.xml");
-    createDirectories(formFile.getParent());
-    Files.copy(sourceFormFile, formFile);
-
-    FormMetadata formMetadata = FormMetadata.from(formFile);
-    InMemoryFormMetadataAdapter formMetadataPort = new InMemoryFormMetadataAdapter();
-    formMetadataPort.persist(formMetadata);
-    BriefcaseFormDefinition localFormDef = BriefcaseFormDefinition.resolveAgainstBriefcaseDefn(formFile.toFile(), false, briefcaseDir.toFile());
-    FormStatus formStatus = new FormStatus(localFormDef);
-
-    ExportConfiguration configuration = new ExportConfiguration(
-        Optional.<String>empty(),
-        Optional.of(exportDir),
-        Optional.<Path>empty(),
-        DateRange.empty(),
-        OverridableBoolean.FALSE,
-        OverridableBoolean.TRUE,
-        OverridableBoolean.TRUE,
-        OverridableBoolean.FALSE,
-        OverridableBoolean.FALSE,
-        OverridableBoolean.FALSE,
-        OverridableBoolean.FALSE
-    );
-    ExportToCsv.export(formMetadataPort, formMetadata, formStatus, FormDefinition.from(localFormDef), briefcaseDir, configuration);
+    FormDefinition formDefinition = FormDefinition.from(getPath("custom-function-form.xml"));
+    // Assert anything about the form definition object
+    // What's important is that we can get the form def object without errors
+    assertThat(formDefinition.getFormId(), Matchers.is("custom-function-form"));
   }
 }

--- a/test/java/org/opendatakit/briefcase/CustomFunctionTest.java
+++ b/test/java/org/opendatakit/briefcase/CustomFunctionTest.java
@@ -1,0 +1,67 @@
+package org.opendatakit.briefcase;
+
+import static org.opendatakit.briefcase.reused.UncheckedFiles.createDirectories;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.createTempDirectory;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.toURI;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.junit.Test;
+import org.opendatakit.briefcase.export.DateRange;
+import org.opendatakit.briefcase.export.ExportConfiguration;
+import org.opendatakit.briefcase.export.ExportToCsv;
+import org.opendatakit.briefcase.export.FormDefinition;
+import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
+import org.opendatakit.briefcase.model.FormStatus;
+import org.opendatakit.briefcase.model.form.FormMetadata;
+import org.opendatakit.briefcase.model.form.InMemoryFormMetadataAdapter;
+import org.opendatakit.briefcase.pull.aggregate.PullFromAggregateIntegrationTest;
+import org.opendatakit.briefcase.reused.OverridableBoolean;
+import org.opendatakit.briefcase.util.BadFormDefinition;
+
+public class CustomFunctionTest {
+  private static Path getPath(String fileName) {
+    return Optional.ofNullable(PullFromAggregateIntegrationTest.class.getClassLoader().getResource("org/opendatakit/briefcase/" + fileName))
+        .map(url -> Paths.get(toURI(url)))
+        .orElseThrow(RuntimeException::new);
+  }
+
+  @Test
+  public void name() {
+    FormDefinition.from(getPath("custom-function-form.xml"));
+  }
+
+  @Test
+  public void full_regression_test() throws IOException, BadFormDefinition {
+    Path exportDir = createTempDirectory("briefcase-test-export-dir-");
+    Path sourceFormFile = getPath("custom-function-form.xml");
+    Path briefcaseDir = createTempDirectory("briefcase-test-storage-dir-");
+    Path formFile = briefcaseDir.resolve("Custom function form").resolve("Custom function form.xml");
+    createDirectories(formFile.getParent());
+    Files.copy(sourceFormFile, formFile);
+
+    FormMetadata formMetadata = FormMetadata.from(formFile);
+    InMemoryFormMetadataAdapter formMetadataPort = new InMemoryFormMetadataAdapter();
+    formMetadataPort.persist(formMetadata);
+    BriefcaseFormDefinition localFormDef = BriefcaseFormDefinition.resolveAgainstBriefcaseDefn(formFile.toFile(), false, briefcaseDir.toFile());
+    FormStatus formStatus = new FormStatus(localFormDef);
+
+    ExportConfiguration configuration = new ExportConfiguration(
+        Optional.<String>empty(),
+        Optional.of(exportDir),
+        Optional.<Path>empty(),
+        DateRange.empty(),
+        OverridableBoolean.FALSE,
+        OverridableBoolean.TRUE,
+        OverridableBoolean.TRUE,
+        OverridableBoolean.FALSE,
+        OverridableBoolean.FALSE,
+        OverridableBoolean.FALSE,
+        OverridableBoolean.FALSE
+    );
+    ExportToCsv.export(formMetadataPort, formMetadata, formStatus, FormDefinition.from(localFormDef), briefcaseDir, configuration);
+  }
+}

--- a/test/resources/org/opendatakit/briefcase/custom-function-form.xml
+++ b/test/resources/org/opendatakit/briefcase/custom-function-form.xml
@@ -1,0 +1,25 @@
+<?xml version="1.0"?>
+<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:h="http://www.w3.org/1999/xhtml">
+  <h:head>
+    <h:title>Custom function form</h:title>
+    <model>
+      <instance>
+        <data id="custom-function-form">
+          <some-field/>
+          <some-field-trimmed/>
+          <meta>
+            <instanceID/>
+          </meta>
+        </data>
+      </instance>
+      <bind nodeset="/data/some-field" type="string"/>
+      <bind nodeset="/data/some-field-trimmed" type="string" calculate="trim(/data/some-field)"/>
+      <bind calculate="concat('uuid:', uuid())" nodeset="/data/meta/instanceID" readonly="true()" type="string"/>
+    </model>
+  </h:head>
+  <h:body>
+    <input ref="/data/some-field">
+      <label>Some field</label>
+    </input>
+  </h:body>
+</h:html>

